### PR TITLE
chore: use in-app state for consistency

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-480](https://linear.app/customerio/issue/MBL-480/update-in-app-messaging-module-using-state-management-in-ios)

### Changes

- Utilized `InAppMessageState` throughout all relevant classes
- Used environment from state class
- Updated `LogManager` and moved it to `DiGraph`
- Added `runMainAfterDelay` to `ThreadUtil` to allow reuse and avoid direct usage of `DispatchQueue`
- Removed `RouteManager` and `UserManager` as they are no longer needed

### CODE REVIEW

File: Sources/MessagingInApp/Gist/Gist.swift
Comment (mrehan27) [inline Sources/MessagingInApp/Gist/Gist.swift:1]: Most of the code in this class is updated in next PRs in the chain, so the code is commented to make tracking changes easier.

File: Sources/MessagingInApp/Gist/Gist.swift [hunk 1]
```diff
@@ -38,41 +37,42 @@ public class Gist: GistDelegate {
...
-        UserManager().setUserToken(userToken: userToken)
+        // UserManager().setUserToken(userToken: userToken)
     }
 
     public func clearUserToken() {
         cancelModalMessage(ifDoesNotMatchRoute: "") // provide a new route to trigger a modal cancel.
         messageQueueManager.clearLocalStore()
-        UserManager().clearUserToken()
+        // UserManager().clearUserToken()
         messageQueueManager.clearUserMessagesFromLocalStore()
```
Comment (Shahroz16) [inline Sources/MessagingInApp/Gist/Gist.swift:53]: does `messageQueueManager` also get removed in later PRs, cause I don't see it commented here?

File: Sources/MessagingInApp/Gist/Gist.swift [hunk 2]
```diff
@@ -140,9 +140,9 @@ public class Gist: GistDelegate {
         if let queueId = message.queueId {
             shownMessageQueueIds.insert(queueId)
         }
-        let userToken = UserManager().getUserToken()
-        LogManager(siteId: siteId, dataCenter: dataCenter)
-            .logView(message: message, userToken: userToken) { response in
+        let state = InAppMessageState()
```
Comment (Shahroz16) [inline Sources/MessagingInApp/Gist/Gist.swift:143]: i dont understand this one, why aren't we using the state from manager?

File: Sources/MessagingInApp/Gist/Managers/MessageManager.swift [hunk 3]
```diff
@@ -33,12 +33,12 @@ class MessageManager: EngineWebDelegate {
             siteId: Gist.shared.siteId,
             dataCenter: Gist.shared.dataCenter,
```
Comment (Shahroz16) [inline Sources/MessagingInApp/Gist/Managers/MessageManager.swift:34]: shouldn't these 2 also fetched from state?

**Review by Shahroz16 — APPROVED** [review]
It was a bit hard to review this not know whats to follow next 😅 major thing I see if InAppMessageState instance being created in classes, which I am guessing will be moved in later PR, if not that's a blocker.

File: Sources/MessagingInApp/Gist/Gist.swift (hunk 1 above)
Comment (mrehan27) [reply Sources/MessagingInApp/Gist/Gist.swift:53]: `messageQueueManager` doesn't get removed, but the call does. You can take a look at PR #802 to see the final implementation.

File: Sources/MessagingInApp/Gist/Gist.swift (hunk 2 above)
Comment (mrehan27) [reply Sources/MessagingInApp/Gist/Gist.swift:143]: This is just to keep things compiling. As I mentioned above, this class is almost completely rewritten in PR #802

File: Sources/MessagingInApp/Gist/Managers/MessageManager.swift (hunk 3 above)
Comment (mrehan27) [reply Sources/MessagingInApp/Gist/Managers/MessageManager.swift:34]: Yup, correct. Updated in #802

**Pull request merged**